### PR TITLE
Feature/update architecture

### PR DESCRIPTION
**Pull request by jenniferboedker**
Updated how data is written to a file. This should not be done by a presenter. 

I moved the code from the presenter to a respective class

### CODE REVIEW

File: metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentDataFileOutput.groovy [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package life.qbic.metastats.filter
+
+import life.qbic.metastats.datamodel.MetaStatsPackageEntry
+
```
Comment (sven1103) [inline metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentDataFileOutput.groovy:4]: Please add interface, author and version information

File: metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentDataFileOutput.groovy [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+interface FilterExperimentDataFileOutput {
+
+
+    /**
+     * Write the created metadata into a file and download it to the user system
+     * @param entries with the metadata information of the project
+     * @return
+     */
+    def writeMetaDataToFile(List<MetaStatsPackageEntry> entries)
```
Comment (sven1103) [inline metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentDataFileOutput.groovy]: I suggest to remove implementation details here. File is a detail.
```suggestion
    def write(List<MetaStatsPackageEntry> entries)
```

File: metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentDataFileOutput.groovy (hunk 1 above)
Comment (jenniferboedker) [reply metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentDataFileOutput.groovy:4]: Do you mean at the top of the interface?

**Review by Steffengreiner — APPROVED** [review]
Looks good!

File: metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentDataFileOutput.groovy (hunk 1 above)
Comment (sven1103) [reply metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentDataFileOutput.groovy:4]: yes exactly, as JavaDoc comment

**Review by sven1103 — CHANGES_REQUESTED** [review]
Some minor things

File: metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentDataFileOutput.groovy (hunk 1 above)
Comment (jenniferboedker) [reply metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentDataFileOutput.groovy:4]: done with 10ca084

File: metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentDataFileOutput.groovy (hunk 2 above)
Comment (jenniferboedker) [reply metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentDataFileOutput.groovy]: 50ecbbc

File: metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentData.groovy [hunk 3]
```diff
@@ -10,7 +10,7 @@ import org.apache.logging.log4j.Logger
  * Use sase to handle filtering of the data loaded by the RequestExperimentData use case
```
Comment (Steffengreiner) [inline metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentData.groovy]:
```suggestion
 * Use case to handle filtering of the data loaded by the RequestExperimentData use case
```

File: metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentData.groovy [hunk 4]
```diff
@@ -10,7 +10,7 @@ import org.apache.logging.log4j.Logger
...
  *
  * This class filters experiment and sample data for the required metadata of the output. The data is verified and forwarded
- * to a output class
+ * to a output class {@link FilterExperimentDataFileOutput}
```
Comment (Steffengreiner) [inline metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentData.groovy]:
```suggestion
 * to an output class {@link FilterExperimentDataFileOutput}
```

File: metastats-infrastructure/src/main/groovy/life/qbic/metastats/MetaStatsPresenter.groovy [hunk 5]
```diff
@@ -6,7 +6,10 @@ import org.apache.logging.log4j.Logger
 /**
  * Controls how data is presented
  *
- * This class responsible for how the data is presented to the user.
+ * This class responsible for how the data is presented to the user. It acts as connector between the use cases and the view
```
Comment (Steffengreiner) [inline metastats-infrastructure/src/main/groovy/life/qbic/metastats/MetaStatsPresenter.groovy]:
```suggestion
 * This class is responsible for how the data is presented to the user. It acts as connector between the use cases and the view
```

File: metastats-infrastructure/src/main/groovy/life/qbic/metastats/MetaStatsPresenter.groovy [hunk 6]
```diff
@@ -6,7 +6,10 @@ import org.apache.logging.log4j.Logger
...
+ * and should be used to transfere data between these components.
```
Comment (Steffengreiner) [inline metastats-infrastructure/src/main/groovy/life/qbic/metastats/MetaStatsPresenter.groovy]:
```suggestion
 * and should be used to transfer data between these components.
```

File: metastats-infrastructure/src/main/groovy/life/qbic/metastats/request/OpenBisParser.groovy [hunk 7]
```diff
@@ -11,7 +11,10 @@ import org.apache.logging.log4j.Logger
  * Handles how OpenBis data is parsed
  *
  * This class is responsible for creating MetaStatsExperiments and MetaStatsSamples from OpenBis experiments and samples.
- *
+ * Based on OpenBis preparation samples it fetches the sample information of the parent and child samples. All this information
+ * is condensed in one MetaStatsSample. This sample will be contain all sample information required for the output.
```
Comment (Steffengreiner) [inline metastats-infrastructure/src/main/groovy/life/qbic/metastats/request/OpenBisParser.groovy:15]:
```suggestion
 * is condensed in one MetaStatsSample. MetaStatsSample will contain all sample information required for the output.
```

File: metastats-infrastructure/src/main/groovy/life/qbic/metastats/request/OpenBisParser.groovy [hunk 8]
```diff
@@ -11,7 +11,10 @@ import org.apache.logging.log4j.Logger
...
+ * If some properties like Q_SECONDARY_NAME are not unique they are masked.
```
Comment (Steffengreiner) [inline metastats-infrastructure/src/main/groovy/life/qbic/metastats/request/OpenBisParser.groovy]:
```suggestion
 * Non unique properties such as Q_SECONDARY_NAME are masked.
```

**Review by Steffengreiner — COMMENTED** [review]
I think the description are fine and explain the individual classes well. I only found some small errors in terms of grammar and language.

**Review by sven1103 — APPROVED** [review]

**Comment by sven1103** [comment]
Please squash and merge

**Comment by jenniferboedker** [comment]
This PR updates the javadoc code documentation of the software, updates the architecture in terms of renaming classes according to their function and updates how the file output is generated.

**Pull request merged**